scripts/user.js

```diff
@@ -282,32 +282,13 @@ function openTripHistory() {
 		const tripListElement = document.createElement("li");
 		tripListElement.className = "trip-list-element";
 
-		// Get formatted date
-		let tripDate = new Date(trip.startDate);
-		let monthNumberToShortForm = [
-			"jan.",
-			"fev.",
-			"mar.",
-			"abr.",
-			"mai.",
-			"jun.",
-			"jul.",
-			"ago.",
-			"set.",
-			"out.",
-			"nov.",
-			"dez.",
-		];
-		const formattedDate = `${tripDate.getDate()} ${
-			monthNumberToShortForm[tripDate.getMonth()]
-		} ${tripDate.getFullYear()}`;
+		// Get formatted date. Format: "1 de jan. de 2024"
+		const formattedDate = tripTime.toLocaleDateString("pt", { dateStyle: "medium" });
```

Comment:
Estavas a tentar aceder ao tripTime antes de ser definido
Pus também o replaceAll para retirar os "de" da data (mais por uma questão do layout)

```suggestion
		const tripDate = new Date(Date.parse(trip.startDate));
		const formattedDate = tripDate.toLocaleDateString("pt", { dateStyle: "medium" }).replaceAll(" de ", " ");
```

Reply:
Neste caso é redundante fazer new Date com Date.parse
```suggestion
		const tripDate = new Date(trip.startDate);
		const formattedDate = tripDate.toLocaleDateString("pt", { dateStyle: "medium" }).replaceAll(" de ", " ");```